pkg/pointerdb/pointerdb.go

```diff
@@ -228,3 +231,70 @@ func (s *Server) Delete(ctx context.Context, req *pb.DeleteRequest) (resp *pb.De
 	s.logger.Debug("deleted pointer at path: " + req.GetPath())
 	return &pb.DeleteResponse{}, nil
 }
+
+func (s *Server) iterate(ctx context.Context, req *pb.IterateRequest, queue dr.Queue) (err error) {
```

Comment:
The primary reason for this is not iteration, the purpose is something else. Try naming it something like `listInDangerPieces`.